Content.Server/Explosion/EntitySystems/GridExplosion.cs

```diff
@@ -0,0 +1,304 @@
+using Content.Shared.Atmos;
+using Robust.Shared.Map;
+using static Content.Server.Explosion.EntitySystems.ExplosionSystem;
+
+namespace Content.Server.Explosion.EntitySystems;
+
+public sealed class GridExplosion : TileExplosion
+{
+    public IMapGrid Grid;
```

Comment:
kind of at a loss as to what these grid/space/tile explosion classes are for, exactly, or why they're separate from the `Explosion` class that also exists, or any of the data structures that are in `ExplosionSystem` itself?

Reply:
Yeah this is a bit of a mess and can probably do with re-organising

So currently there is:
- an `Explosion` class in the client overlay system, which is just for storing the data received from the server about how/where to draw explosions
- an `Explosion` class in the server's ExplosionSystem, which is used to store explosion data for processing over several ticks. Has tile enumerators & keeps track of entities that have already been damaged by the explosion to avoid double damaging.

Those two are probably fine, though maybe they need a more descriptive or at least a distinct name. They basically just exist to hold data (and for the server one, enumerate over tiles).

The other two classes `GridExplosion` & `SpaceExplosion`, which inherit from `TileExplosion`, exist almost exclusively for the `GetExplosionTiles()` function, and could almost just be local to the function if C# somehow supported that.
The each of those classes basically have information about the tiles on a given grid (or space), and have a function that performs the neighbour finding steps to add new tiles. So you can basically create one and then iteratively expand the area.

Originally, back when explosions only worked on a single grid, the `GetExplosionTiles()` was basically just several large function that called each other with 10+ arguments, which was quite a mess. Then I made them all local-functions using function local variables to make it slightly cleaner. Then after multi-grid & space explosions I basically just split the local variables and functions inside `GetExplosionTiles()` into separate classes. So thats why they are the way they are, but can probably be cleaned up or made nicer somehow. I will throw in some more comments.